modelgym/xgboost_model.py

```diff
@@ -51,11 +51,25 @@ def preprocess_params(self, params):
             params['max_depth'] = int(params['max_depth'])
         return params
 
-    def set_params(self, **kwargs):
+    def set_parameters(self, **kwargs):
         for key in kwargs:
             dic = {key: kwargs.get(key)}
             self.default_params.update(dic)
 
+    def set_parameter(self, params):
```

Comment:
I'd joint it with `set_parameters` which would turn into `set_parameters(self, *args, **kwargs)` for all models